marathon/src/test/scala/io/buoyant/marathon/v2/AppIdNamerTest.scala

```diff
@@ -235,37 +245,40 @@ class AppIdNamerTest extends FunSuite with Awaits {
       initAddrsAlive = true
     )
 
-    val namer = new AppIdNamer(api, Path.Utf8("io.l5d.marathon"), ttl)
+    val namer = new AppIdNamer(api, Path.Utf8("io.l5d.marathon"), ttl, timer)
     @volatile var state: Activity.State[NameTree[Name]] = Activity.Pending
-    namer.lookup(Path.read("/bar/residual")).states.respond(state = _)
-    assert(state == Activity.Ok(NameTree.Leaf(Path.Utf8("io.l5d.marathon", "bar"))))
 
-    @volatile var boundAddr: Var[Addr] = Var.value(Addr.Neg)
-    state match {
-      case Activity.Ok(NameTree.Leaf(bound: Name.Bound)) =>
-        boundAddr = bound.addr
-      case state => fail(s"$state is not a NameTree.Leaf[Name]")
-    }
+    Time.withCurrentTimeFrozen { tc =>
+      namer.lookup(Path.read("/bar/residual")).states.respond(state = _)
+      assert(state == Activity.Ok(NameTree.Leaf(Path.Utf8("io.l5d.marathon", "bar"))))
 
-    @volatile var addrTest = Addr.Bound(Address(0))
-    val closable = boundAddr.changes.respond(addrTest = _)
-    assert(addrTest == Addr.Bound(addr))
+      @volatile var boundAddr: Var[Addr] = Var.value(Addr.Neg)
+      state match {
+        case Activity.Ok(NameTree.Leaf(bound: Name.Bound)) =>
+          boundAddr = bound.addr
+        case state => fail(s"$state is not a NameTree.Leaf[Name]")
+      }
 
-    assertOnAddrs(state) { addrs =>
-      assert(addrs.size == 1)
-      assert(addrs.head == addr)
-    }
+      @volatile var addrTest = Addr.Bound(Address(0))
+      val closable = boundAddr.changes.respond(addrTest = _)
+      assert(addrTest == Addr.Bound(addr))
 
-    // wait for marathon api to go down
-    api.addrsAlive = false
-    Thread.sleep(ttl.inMillis * 3)
+      assertOnAddrs(state) { addrs =>
+        assert(addrs.size == 1)
+        assert(addrs.head == addr)
+      }
 
-    namer.lookup(Path.read("/bar/residual")).states.respond(state = _)
-    assertOnAddrs(state) { addrs =>
-      assert(addrs.size == 1)
-      assert(addrs.head == addr)
-    }
+      api.addrsAlive = false
+      tc.advance(ttl)
+      timer.tick()
+
+      namer.lookup(Path.read("/bar/residual")).states.respond(state = _)
+      assertOnAddrs(state) { addrs =>
+        assert(addrs.size == 1)
+        assert(addrs.head == addr)
```

Comment:
i'd probably just `assert(addrs == Set(addr))` -- scalatest is generally good at turning that into a reasonable error.